src/main.rs

```diff
@@ -0,0 +1,113 @@
+use clap::Parser;
+use serde_json::{Result, Value};
+use std::{fs::File, io::BufReader};
+
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+struct Args {
+    #[arg(short, long)]
+    file: String,
+
+    #[arg(short, long)]
+    output: String,
+
+    #[arg(short, long, action)]
+    pretty: bool,
+}
+
+fn main() {
+    env_logger::init();
+    let args = Args::parse();
+
+    log::debug!(
+        "Attempting read from {}, and will try write to {}",
+        args.file,
+        args.output
+    );
+
+    let json_data = &mut read_from_file(args.file).expect("Failed to parse the file contents!");
```

Comment:
This is debatable, but you may change signature of main to `fn main() -> anyhow::Result<()>` instead of panic.

Reply:
Simply speaking - yes. This is a binary file, so it's fine to panic here.